mediachain/transactor/blockchain_follower.py

```diff
@@ -26,11 +26,13 @@ def __init__(self,
         self.stream_func = stream_func
         self.block_ref_queue = Queue()
         self.incoming_event_queue = Queue()
-        self.caught_up = False
+        self.block_replay_queue = Queue()
```

Comment:
We shouldn't OOM, since the `block_replay_queue` just has block references in it, not the blocks themselves.

The basic flow goes:

- receive ref to current block, put on `block_ref_queue`
- catchup thread fetches block and puts the ref onto `block_replay_queue`. also puts the block's `chain` ref onto `block_ref_queue` so the catchup thread will fetch the previous block
- the `_event_stream` iterator waits for the `catchup_complete` signal, then pulls the refs off of the `block_replay_queue` and fetches the blocks.  This is fast b/c they were cached previously.  It replays all the events from each block and then plays events from the `incoming_event_queue`

dammit; describing that made me realize that I need to be using a LIFO queue for the replay.  I was using a dequeue before and appending to the left.  with the FIFO queue in there now, it'll play the blocks back in the wrong order.